src/main/java/me/moodcat/backend/rooms/RoomInstance.java

```diff
@@ -206,8 +217,11 @@ private void processVotes(final Song previousSong) {
     private void processNextSong(final Room room, final List<Song> history) {
         final List<Song> playQueue = room.getPlayQueue();
         if (playQueue.isEmpty()) {
-            playQueue.addAll(history);
-            history.clear();
+            VAVector vector = this.roomDAOProvider.get().findById(id).getVaVector();
+            playQueue.addAll(this.songDAOProvider.get().findForDistance(vector, NUMBER_OF_SELECTED_SONGS));
+            
+            int historyLength = history.size();
+            room.setPlayHistory(history.subList(Math.max(0, historyLength - NUMBER_OF_SELECTED_SONGS), historyLength));
```

Comment:
I think this statement is wrong. It's kindof unreadable imo.
Lets write it like this:
```java
// Math.max(0, length) for clamping >= 0.
int historySize = history.size();
int firstIndex = Math.max(0, historySize - NUMBER_OF_SELECTED_SONGS);
int lastIndex = historySize - 1; // I think this should be -1 because we're talking about indexes, not length?
room.setPlayHistory(history.subList(firstIndex, lastIndex));
```